core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -177,6 +178,43 @@ public synchronized boolean isTerminated() {
   }
 
   /**
+   * Ensures that there is an active transport and that it is started.
+   */
+  public void tryStartTransport() {
+    obtainActiveTransport();
+  }
+
+  /**
+   * Shuts down the current transport and starts a new one. If there is no active transport, then
+   * starts one. This can be useful to force the transport to connect to a different backend when
+   * the remote endpoint is a load balancer.
+   */
+  public synchronized void recreateTransport() {
+    if (activeTransport != null) {
+      activeTransport.shutdown();
+      activeTransport = null;
+    }
+    tryStartTransport();
+  }
+
+  /**
+   * Pings the remote endpoint to verify that the transport is still active. The future will
+   * complete when the remote endpoint replies. The future completes exceptionally if the transport
+   * is shutdown before the response is received. The value of the future is the round-trip time of
+   * the ping, in nanoseconds.
+   *
+   * <p>Transports that do not have any mechanism by which to ping the remote endpoint should
+   * return {@link com.google.common.util.concurrent.Futures#immediateFuture(Object)
+   * immediateFuture(null)}.
+   *
+   * @return a future that completes when the remote endpoint responds to the ping with a value of
+   *     the round-trip time in nanoseconds
+   */
+  public ListenableFuture<Long> ping() {
```

Comment:
Yeah I think it's probably worth while to avoid guava if possible.  Maybe something like:

```java
...
void ping(PongListener listener);
...
interface PongListener {
    void onPong(long rttMillis);
}
```
@ejona86 WDYT?